### PR TITLE
Meltdown-DE for AMD.

### PR DESCRIPTION
**Pull request by asteinha**
Tested on AMD. Does not work on Intel.
I'll factor out those signal handlers into utils after #54 is merged.

### CODE REVIEW

File: demos/meltdown_de.cc [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+ * 8 % 0 = 2
+ * 9 % 0 = 2
+ * 10 % 0 = 2
+ * 11 % 0 = 2
+ * 12 % 0 = 3
+ * 13 % 0 = 3
+ * 14 % 0 = 3
+ * 15 % 0 = 3
+ * 16 % 0 = 2
+ * 7 times more 2
```
Comment (mmdriley) [inline demos/meltdown_de.cc]: Not sure how to read this comment. Does this mean that `x % 0 == 2` for x \in {17..23}?

File: demos/meltdown_de.cc [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+#  error Unsupported OS. Linux required.
+#endif
+
+#if !SAFESIDE_IA32 && !SAFESIDE_X64
+#  error Unsupported architecture. AMD required.
+#endif
+
+#include <array>
+#include <cstring>
+#include <fstream>
```
Comment (mmdriley) [inline demos/meltdown_de.cc]: unused I think

File: demos/meltdown_de.cc [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+  "XXc",
+  "XXr",
+  "XXe",
+  "XXt",
+  "XX!",
+  "XX!",
+  "XX!",
+};
+
+constexpr size_t kPrivateDataLength = 16;
```
Comment (mmdriley) [inline demos/meltdown_de.cc]: let's either compute this with `sizeof(private_data)/sizeof(*private_data)` or, maybe better, put the length definition first and use it in the array declaration: `const char *private_data[kPrivateDataLength] = ...`

**Review by mmdriley — APPROVED** [review]

File: demos/meltdown_de.cc (hunk 1 above)
Comment (asteinha) [reply demos/meltdown_de.cc]: Yes.

Comment (asteinha) [reply demos/meltdown_de.cc]: Extended the comment.

File: demos/meltdown_de.cc (hunk 2 above)
Comment (asteinha) [reply demos/meltdown_de.cc]: Removed.

File: demos/meltdown_de.cc (hunk 3 above)
Comment (asteinha) [reply demos/meltdown_de.cc]: Done.

**Pull request merged**